src/InterventionBackend.php

```diff
@@ -371,7 +304,7 @@ public function writeToStore(AssetStore $assetStore, $filename, $hash = null, $v
             $extension = pathinfo($url, PATHINFO_EXTENSION);
             // Save file
             $result = $assetStore->setFromString(
-                $resource->encode($extension, $this->getQuality())->getEncoded(),
+                $resource->encodeByExtension($extension, $this->getQuality())->toString(),
```

Comment:
Upgrade guide doesn't mention this (except to say `encode()` has a new signature) but this seems like the easiest way to keep the existing behaviour here.